### PR TITLE
Storage explorer improvements for large folders

### PR DESCRIPTION
**Pull request by inian**
- reduces batch size for move and delete to 10. This is slower but we dont run into rate limits and reduces the number of 502s we get from the backend. 
- Deletes and Moves are best effort. One file not being moved to the new folder doesn't block the entire move operation. 
- Only the first 200 files in each folder were being moved previously. Now all files are moved / deleted.

With this PR, I was able to upload, rename and delete a folder with 14k files without any issue.

### CODE REVIEW

File: studio/localStores/storageExplorer/StorageExplorerStore.js [hunk 1]
```diff
@@ -1126,11 +1130,14 @@ class StorageExplorerStore {
...
         return () => {
-          return new Promise(async (resolve, reject) => {
+          return new Promise(async (resolve) => {
             const { error } = await this.supabaseClient.storage
               .from(this.selectedBucket.name)
               .move(fromPath, toPath)
-            if (error) reject()
+              if (error) {
+                hasErrors = true
+                toast.error(`Failed to move ${fromPath} to the new folder`)
```
Comment (darora) [inline studio/localStores/storageExplorer/StorageExplorerStore.js:1139]: think you're supposed to use ui.setNotification ? https://github.com/supabase/supabase/pull/7348/files

Comment (inian) [reply studio/localStores/storageExplorer/StorageExplorerStore.js:1139]: having troubling importing this the correct way. @joshenlim lmk if i can pair with you on this

Comment (joshenlim) [reply studio/localStores/storageExplorer/StorageExplorerStore.js:1139]: no worries this is fine for now - i can help to clean it up in another PR too, this whole file needs some clean up to use the ui store anyways

**Review by joshenlim — APPROVED** [review]
lgtm! thanks so much 🙏

**Pull request merged**